### PR TITLE
Checkstyle: Adds another system default check

### PR DESCRIPTION
**Pull request by adamvoncorswant**
`Year.now()` should not be used without a zone id

@magnuslambert or @danielefoerberghi perhaps?

### CODE REVIEW

File: checkstyle/src/main/resources/checkstyle/checkstyle-without-suppression.xml [hunk 1]
```diff
@@ -30,6 +30,11 @@
             <property name="message" value="Inappropriate usage of ZonedDateTime.now(). Time time zone should be specified rather than using system default."/>
             <property name="ignoreComments" value="true"/>
         </module>
+        <module name="RegexpSinglelineJava">
+            <property name="format" value="Year\.now\(\)"/>
```
[the remark this answers is not in the document]
Comment (adamvoncorswant) [reply checkstyle/src/main/resources/checkstyle/checkstyle-without-suppression.xml]: Yes, this is definitely not an exhaustive check. It suffers mainly from being just dumb text filters. That said, `Clock.systemDefaultZone()` is already checked for (further down), and the second: `Year.of(2018)` is perfectly legit, don't you think? :smile:

[the remark this answers is not in the document]
Comment (adamvoncorswant) [reply checkstyle/src/main/resources/checkstyle/checkstyle-without-suppression.xml]: Can the confusion be that the naming of `Year` is inconsistent with the other `local` time classes, and should really have been named `LocalYear`?

[the remark this answers is not in the document]
Comment (magnuslambert) [reply checkstyle/src/main/resources/checkstyle/checkstyle-without-suppression.xml]: Yes, sorry my bad. Looked up the source code in java.time, but read the wrong lines :)

**Comment by adamvoncorswant** [comment]
Thanks @magnuslambert !

**Pull request merged**